src/connectionController.ts

```diff
@@ -313,20 +325,19 @@ export default class ConnectionController {
   public async removeMongoDBConnection(connectionId: string): Promise<boolean> {
     // Ensure we aren't currently connecting.
     if (this._connecting) {
-      return Promise.reject(
-        new Error('Unable to remove connection: currently connecting.')
-      );
+      vscode.window.showErrorMessage('Unable to remove connection: currently connecting.');
+      return Promise.resolve(false);
     }
     // Ensure we aren't currently disconnecting.
     if (this._disconnecting) {
-      return Promise.reject(
-        new Error('Unable to remove connection: currently disconnecting.')
-      );
+      vscode.window.showErrorMessage('Unable to remove connection: currently disconnecting.');
+      return Promise.resolve(false);
     }
 
     if (!this._connectionConfigs[connectionId]) {
       // No active connection(s) to remove.
-      return Promise.reject(new Error('Connection does not exist.'));
+      vscode.window.showErrorMessage('Connection does not exist.');
+      return Promise.resolve(false);
```

Comment:
Here's an example of where instead of `reject`ing the promise and letting vscode handle the message displaying, we're doing it ourselves to make the messages a bit clearer.

Reply:
Do you have screens of how the old error looked like and how the new one looks now? I run this branch and can't notice the difference... :)

Reply:
Hmmm interesting, I tried myself. It appears that the error messaging is bad only when using commands originating from the tree view. Here's an example of what those look like:
![Screen Shot 2020-03-09 at 1 40 27 PM](https://user-images.githubusercontent.com/1791149/76214193-83e25800-620c-11ea-9284-d75b1e405bf8.png)

Interestingly though, commands from the command prompt just display the error message. Since some of these methods are used in command handlers from both the command prompt and tree view I think we want to probably want to go with `vscode.window.showErrorMessage('Connection does not exist.')` and `resolve(false)`